pkg/webhook/controlplane/genericmutator/mutator.go

```diff
@@ -102,6 +108,39 @@ func (m *mutator) InjectClient(client client.Client) error {
 	return nil
 }
 
+type ensurerContext struct {
+	client  client.Client
+	object  metav1.Object
+	cluster *extensionscontroller.Cluster
+}
+
+// NewEnsurerContext creates a ensurer context object.
+func NewEnsurerContext(client client.Client, object metav1.Object) EnsurerContext {
+	return &ensurerContext{
+		client: client,
+		object: object,
+	}
+}
+
+// NewInternalEnsurerContext creates a ensurer context object.
+func NewInternalEnsurerContext(cluster *extensionscontroller.Cluster) EnsurerContext {
+	return &ensurerContext{
+		cluster: cluster,
+	}
+}
+
+// GetCluster returns the cluster object.
+func (c *ensurerContext) GetCluster(ctx context.Context) (*extensionscontroller.Cluster, error) {
```

Comment:
Why don't we simply fetch it always and inject it into the ensurer context by default? Then the various webhook implementors don't need to call this function and handle the error, etc. WDYT?

Reply:
We wanted to lazy load the `cluster` object as needed.